### DISCUSSION TITLE
Installation of XQuartz Not Found

### DISCUSSION THREAD

**Discussion by ries9112** [community/community#25418] [Actions]
<p>Hello, I am running a workflow on GitHub Actions and I am looking to run some R code using a package called “rayshader” which requires something called “XQuartz” to be installed.</p>
<p>I am able to install XQuartz successfully using the command:</p>
<pre><code class="lang-auto">brew cask install xquartz
</code></pre>
<p>The install seems to work based on the results I’m seeing:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/c/9/c90b77bfdde996db682490ca1d4d0fa5a974521f.png" data-download-href="/uploads/short-url/sGwxyawAZhiC2VA54wh9IC0lO9h.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/32306488/181091021-e3b2c6ba-336a-4ab4-882f-3142c0ce28b4.png" alt="image" data-base62-sha1="sGwxyawAZhiC2VA54wh9IC0lO9h" width="517" height="352" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/9/c90b77bfdde996db682490ca1d4d0fa5a974521f_2_517x352.png, https://aws1.discourse-cdn.com/github/optimized/3X/c/9/c90b77bfdde996db682490ca1d4d0fa5a974521f_2_775x528.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/c/9/c90b77bfdde996db682490ca1d4d0fa5a974521f.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/9/c90b77bfdde996db682490ca1d4d0fa5a974521f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">897×612 71.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>But when the code gets to the step that needs XQuartz, I am getting the following error:</p>
<pre><code class="lang-auto">error: X11 not found; XQuartz (from www.xquartz.org) is required to run rgl.
</code></pre>
<p>Thoughts on why that may be the case? I have a similar step where I need to install pandoc through brew and that works as expected and is found when I run the R command that requires it later on (removed from code for simplicity but that one does work as expected when running my R code and is found). You can find the run from the screenshots above here: <a href="https://github.com/ries9112/XQuartz-Test/runs/1306009115?check_suite_focus=true" rel="noopener nofollow ugc">https://github.com/ries9112/XQuartz-Test/runs/1306009115?check_suite_focus=true</a></p>
<p>Here is a simplified .yml file that runs the workflow and reproduces my issue:</p>
<pre><code class="lang-auto">name: Run Tutorial
on: [push]
      
jobs:
  build:
    runs-on: macOS-10.15
    steps:
      - uses: actions/checkout@v2
      - uses: r-lib/actions/setup-r@v1
      - name: Install XQuartz for rayshader
        run: |
          brew cask install xquartz
      - name: Install Packages
        run: |-
          Rscript -e "install.packages('devtools')"
          Rscript -e "devtools::install_github('tylermorganwall/rayshader')"
</code></pre>
<p>Any thoughts on getting this to work? Any help would be much appreciated!</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="ries9112" data-post="1" data-topic="139804">
<div class="title">
<div class="quote-controls"></div>
 ries9112:</div>
<blockquote>
<p><code>error: X11 not found</code></p>
</blockquote>
</aside>
<p>This suggests that the tool needs a graphical environment to run, which isn’t present on the Github hosted runners. On Linux I’d probably use <a href="https://xpra.org/trac/wiki/Xdummy">Xdummy</a> or a VNC server to provide one, I’m not sure what options you have on macOS.</p>

**Comment by ries9112** [comment]
<p>I guess what I’m looking to do isn’t currently possible, that’s unfortunate but makes sense, thank you for the response. Will accept it as the solution unless someone has any other ideas.</p>
